docs/openapi.md

```diff
@@ -1,36 +1,251 @@
+
+OpenAPI [paths](https://github.com/OAI/OpenAPI-Specification/blob/main/versions/3.0.2.md#pathsObject) object provides mapping rules *Verb* and *Pattern* properties. 3scale methods will be associated accordingly to the [operationId](https://github.com/OAI/OpenAPI-Specification/blob/main/versions/3.0.2.md#operationObject)
+
+*Delta* value is hard-coded to `1`.
+
+By default, *Strict matching* policy is being configured. Matching policy can be switched to **Prefix matching** with the `--prefix-matching` command option.
+
+#### Authentication
+
+Just one top level security requirement supported.
+Operation level security requirements are not supported.
+
+Supported security schemes: `apiKey`, `oauth2` (any flow type).
+
+For the `apiKey` security scheme type:
+* *credentials location* will be read from the OpenAPI [in](https://github.com/OAI/OpenAPI-Specification/blob/main/versions/3.0.2.md#security-scheme-object) field of the security scheme object.
+* *Auth user key* will be read from the OpenAPI [name](https://github.com/OAI/OpenAPI-Specification/blob/main/versions/3.0.2.md#security-scheme-object) field of the security scheme object.
+
+Partial example of OpenAPI (3.0.2) with `apiKey` security requirement
+
+```yaml
+---
+openapi: "3.0.2"
+security:
+  - petstore_api_key: []
+components:
+  securitySchemes:
+    petstore_api_key:
+      type: apiKey
+      name: api_key
+      in: header
+```
+
+For the `oauth2` security scheme type:
+* *credentials location* is hard-coded to `headers`.
+* *OpenID Connect Issuer Type* defaults to `rest` and can be overridden using this `--oidc-issuer-type=<value>` command option.
+* *OpenID Connect Issuer* is not read from OpenAPI. Since 3scale requires that the issuer URL must include a *client secret*, the issue must be set using this `--oidc-issuer-endpoint=<value>` command option.
+* *OIDC AUTHORIZATION FLOW* is read from the [flows](https://github.com/OAI/OpenAPI-Specification/blob/main/versions/3.0.2.md#security-scheme-object) field of the security scheme object.
+
+Partial example of OpenAPI (3.0.2) with `oauth2` security requirement
+
+```yaml
+---
+openapi: "3.0.2"
+security:
+    - petstore_oauth:
+      - write:pets
+      - read:pets
+  components:
+    securitySchemes:
+      petstore_oauth:
+        type: oauth2
+        flows:
+          clientCredentials:
+            tokenUrl: http://example.org/api/oauth/dialog
+            scopes:
+              write:pets: modify pets in your account
+              read:pets: read your pets
+```
+
+When OpenAPI does not specify any security requirements:
+* The product is considered as an "Open API"
+* `default_credentials` 3scale policy will be added (also called as `anonymous_policy`)
+* The command option `--default-credentials-userkey` is required and the command will fail if not provided.
+
+#### ActiveDocs
+
+A 3scale ActiveDoc will be created (or updated if previously existed).
```

Comment:
Does that mean that if it previously existed it can alter it in some way? is this controllable? what can potentially differ?

Reply:
For example, when you add/delete/update an operation, the activedoc object is updated accordingly.

Reply:
:+1: